common/network/components/broadcaster.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __COMMON_NETWORK_COMPONENTS_BROADCASTER_H__
+#define __COMMON_NETWORK_COMPONENTS_BROADCASTER_H__
+
+#include "common/network/queues/concurrent_queue_trit_array.h"
+
+typedef concurrent_queue_of_trit_array_p broadcaster_queue_t;
+
+void broadcaster_on_next(broadcaster_queue_t *const queue,
+                         trit_array_t const *const hash);
+void *broadcaster_routine(void *arg);
```

Comment:
all these routines/on_next should be given a `broadcaster_state_t` or similar context object so that we can run multiple in parallel if necessary

Reply:
arg would've been that, it was just to satisfy pthread_create requirements without warning.
But yes, I'll declare some state structs

Reply:
Ah :) fair enough. we can just cast the `pthread_create` part though I guess.
`on_next` will need a state though I guess.